%*******************************************************
% Abstract
%*******************************************************
%\renewcommand{\abstractname}{Abstract}
\pdfbookmark[1]{Abstract}{Abstract}
\begingroup
\let\clearpage\relax
\let\cleardoublepage\relax
\let\cleardoublepage\relax

\chapter*{Abstract}

This thesis revolves around such single cell gene expression datasets in the context of the marine annelid or ragworm \emph{Platynereis dumerilii}, an important model organism, part of the lophotrochozoan taxon of the bilaterians. After describing single cell expression data acquired from Wholemount in Situ Hybridization assays for 169 genes as well as single cell RNA-seq data for 72 cells in the developing brain 48 hours post fertilization of \emph{P. dumerilii}, I discuss the main advantages of both methods and propose a back-mapping method to generate a spatially referenced data set of whole transcriptomes at the single cell level.\\

As the spatial characteristics of the data are crucial to the work presented in this thesis, I also present a 3-dimensional visualization tool that facilitates greatly the upstream and downstream analysis of such datasets. The rest of the thesis focuses on answering the question of identifying cell types from single cell expression data, that is clustering cells together in a meaningful, functional way. Specifically, to take advantage of both the cells location within the tissue and the pattern of gene expression within each cell, I propose a statistical method based on Hidden Markov random fields to cluster the cells according to their gene expression patterns as well as their spatial localization. The method is validated by a simulation study and the quality of the results are compared to those of other clustering methods. Finally the method's output when applied to the \emph{P. dumerilii} in-situ hybridization dataset are biologically validated and functional hypotheses about putative unstudied regions of the brain and their function are formulated.



\vfill



\endgroup			

\vfill