%************************************************
\chapter{Conclusions and future work}\label{ch:conclusions}
%************************************************

\section{Conclusions}

	\subsection{Summary}
	
Throughout this thesis I tried to highlight my main original contributions while presenting work from collaborators and other groups to provide enough context to understand how my work is part of a global effort to investigate spatially referenced gene expression patterns at the single cell level in complex tissues.\\

At the time of writing, the field of transciptomics undergoes major evolutions every year with sequencing methods ever more accurate requiring less and less starting materials. However so far no gene expression assay has been able to achieve single cell resolution of the whole transcriptome while at the same time keeping the spatial organization of the tissue intact. On the one hand, in-situ hybridization succeeds for the spatial component but is not suitable to generate full transcriptomes while, on the other hand single cell RNA-seq can provide with full single cell transcriptomes but no spatial reference.\\

My first original contribution presented in Chapter \ref{ch:singlecell} alongside some methods to process single cell gene expression data, has been to propose a way to combine those two methods in order to create spatially referenced full transcriptomes by mapping single cell RNA-seq results onto a spatial gene expression library generated via in-situ hybridization assays. I have demonstrated the efficiency of the method on a preliminary dataset of single cell RNA-seq data in the brain of \platy{}. This is very recent, and an ongoing effort with my collaborators. However, so far I have been able to map around 75\% of sequenced single cells to a restricted area in the brain defined by the in-situ data. At the time of writing, a manuscript to submit this work as a research paper is being prepared.\\

I have presented in Chapter \ref{ch:non_spatial_clustering_visualization} some background about clustering single cells based on their gene expression patterns and laid the question of visualization to analyse spatially referenced data in a 3D tissue. To answer this question I have presented a software tool that I have developed: ``bioweb3D''. This tool enables scientists who are not visualization experts to easily visualize 3 dimensional data in their web browser without the need to install any software on their computer or to send any data to a remote server. This tool however fairly simple has been of great use for my own work and as a results multiple Figures (\ref{fig:cell_localization}, \ref{fig:eyes_muscles}, \ref{fig:muscles}) in this thesis are screen shots of ``bioweb3D''. I was also pleased following publication of this tool in \emph{Bioinformatics} \cite{Pettit13} to see a few users (around 20 each month) regularly visiting and using ``bioweb3D''.\\

The questions about limitations of single cell expression data as well as the visualization of 3D tissues answered in Chapter \ref{ch:singlecell} and Chapter \ref{ch:non_spatial_clustering_visualization}, I have introduced in Chapter \ref{ch:HMRF}, what has been my main contribution to the field of transcriptomics during the time of my PhD. The HMRF clustering method I have adapted and extended and applied to the single cell gene expression data, was chosen to answer the question of defining cell types from a bottom up approach at the scale of a very complex tissue where the number of cell types is unknown. The method's main advantage is to account for the spatial characteristics of the sites in order to favour spatially smooth clustering results. Not only does this spatial smoothness help the visual downstream analysis, I also hypothesize that considering cells that are spatially close from one another more likely to belong to the same cell type as biologically relevant and potentially important to reduce the impact of experimental noise on the clustering results.\\

 The HMRF method and specifically the impact of the spatial component on the clustering was evaluated and there were good indication that the approach improves the overall quality of clustering as demonstrated by the simulation study detailed in Chapter \ref{ch:simulations}. Indeed, when compared to other non spatial clustering method suitable for single cell gene expression datasets, the HMRF method performs consistently better, which in my opinion justifies the long work put into applying and extending this image analysis method to cluster a completely different type of data. The method was also assessed in term of computational burden, and although legends of the perfect clustering method may circulate, they are unfortunately unfounded. In the case of the HMRF, the price of the improved clustering quality compared to non spatial methods is paid with an exponentially increasing computing time with regard to the number of clusters, which limits this approach to datasets where the true number of clusters is less than 40 when clustering tens of thousands of sites for around a hundred genes. \\
 
 The last chapter (\ref{ch:biology}) details the results when I applied the method to the spatially referenced single cell gene expression data from the brain of \platyfull{}. The outcome was good and in particular, I was able to localize well studied structures of the brain in order to validate the method biologically. Furthermore, I described how some clusters localized in poorly documented regions of the brain could be analysed by taking advantage of the model's final parameters. Indeed, I detailed how a specificity score for each gene and each cluster was computed in order to extract the most specific genes to each cluster. This allowed me to characterize a previously unstudied area of the brain and to formulate an hypothesize about the function of its cells. \\
 
  Even with these satisfactory results there is still work to be done for the clustering method and the single cell RNA-seq back mapping described in Chapter \ref{ch:singlecell}.\\
  
\section{Future work}
  
  \subsection{HMRF future developments}
  The main areas where the HMRF method needs to be improved is in terms of adapting the emission model described in Chapter \ref{ch:HMRF} to whole transcriptomes and quantitative gene expression data. Indeed, as explained in Chapter \ref{ch:singlecell}, my work has been focused in this thesis on clustering binarized data for 89 very well key developmental genes.\\
  
  With an hypothetical dataset in the brain of \platy{} containing the quantitative expression levels for whole transcriptome (more than 10.000 genes) of each cells, how would such a dataset impact the theoretical framework described in Chapter \ref{ch:HMRF}?\\
  
  The quantitative aspect of the data would require a modification of the emission model. Indeed if Bernoulli distributions for each gene in each cluster is suited binarized data, for quantitative data it would be sensible to use Poisson or negative binomial distribution, both approaches used to model gene expression in the literature \todo{ref Poisson negbinom}.\\
  
  The fact that instead of a few selected genes, cells have to be clustered by their whole transcriptome is represents a theoretical issue because genes are not independently expressed. Indeed, as explained in the Introduction (\ref{ch:background}), gene expression is a highly regulated mechanism exhibiting high inter-dependence between genes. More particularly some expressed genes code for regulatory factors that will influence the level of expression of other genes. This inter-dependency is in plain contradiction with the conditional independence hypothesis for gene expression used in the HMRF model.\\
  
  This issue may be resolved by developing methods to analyse whole transcriptomes upstream of the clustering to automatically select an ensemble of genes that are at the same time independent and extremely representative of the overall expression patterns in the studied tissue.\\
  
  Another possible way to deal with this problem would be to try to extend Markov random fields without the conditional independence hypothesis. However, I will not be the judge of the theoretical feasibility of this idea.\\
  
   Of course maybe such an approach won't be needed as it is possible that the errors introduced by some genes inter-dependence will represent a negligible bias compared to the signal brought by thousands of genes. Simulation studies may be able to decide this issue. 


  \subsection{Single cell RNA-seq back-mapping}
  The single cell back-mapping method detailed in Chapter \ref{ch:singlecell}, is the most recent work presented in this thesis. It s an ongoing project with new data available every month. Consequently, the results shown are preliminary and I expect that the method will be developed much further over the next few months. Particularly, a combinatorial approach to the selection of the most specific genes used to do the spatial mapping is currently being investigated.



	



%*****************************************
%*****************************************
%*****************************************
%*****************************************
%*****************************************
