%*****************************************
\chapter{From tissue to single cell transcriptomics, a paradigm shift}\label{ch:singlecell}
%*****************************************
\section{Spatially referenced single cell-like in-situ hybridization data}\label{sec:single_cell_insitu}
  \subsection{Dividing images into "cells"}
  Because in-situ hybridization keeps the studied tissue spatially untouched, achieving single cell gene expression resolution from one image obtained through fluorescent microscopy is a matter of microscope performance and cell size. For big enough cells, single cell resolution has been documented as far as 1989 \cite{tautz89} with some work specifically directed towards achieving this single cell resolution \cite{poulsen93}.\\
  
  When considering \cite{Tomer10} dataset, with current microscope technology, achieving single cell level resolution in \platy{}'s brain on one particular image is feasible. However, our main limitation is the quantity of data involved, indeed, each brain is separated into 20 slices, for 169 genes. This technical bottleneck can be overcome with an automated way of analysing the fluorescence images. However this is not an easy task, as the computer program required needs to be able to ``see" and divide the global picture into cells. Considering that all cells do not exhibit the same shape and size, constructing this ``cell model" is a very complicated task.\\
  
  It is for instance possible to highlight the limits of the cells and to automatically acquire those boundaries through computer vision methods. This process relies on targeting proteins in the membrane or in the extracellular matrix of the cells with specific fluorescent probes. Once the boundaries are acquired, defining every cell is a matter of finding enclosed spaces. To that end, numerous contour detection algorithms exist \cite{li95,fan01,arbelaez11}.\\
  
  Unfortunately, a dataset with the cells limits highlighted does not yet exist for \platy{}'s brain, making a precise division of the images into cells very difficult. Instead, Tomer used a basic approach to divided the images, the ``cube" model \cite{Tomer10}.


  \subsection{A simple cell model, the "cube" data}
  
  Every slice of \platy{}'s brain being aligned onto the reference gene scaffold (see section \ref{sec:gene_expression_lab}) for all 169 genes, the ``cube" model simply consists in dividing each image into squares approximately the size of an average cell. In our dataset, the size chosen was 3 \microm{2}. Importantly, this is actually smaller than the average cell size in \platy{}'s brain. each slice of the brain being approximately 3 \microm{} thick, the resulting dataset, referenced on a 3-dimensional axis, will contain 3 \microm{3} cubes, each of which is attached to the luminescence data for 169 genes.\\
  
  Of course this cell model is far from perfect, it assumes that every cell in the brain is roughly the same size and cubical, which is clearly not the case. Consequently, the ``cube" model will introduce errors in the dataset. The first type of error occurs within areas where the genes under study are highly expressed. In that case, the florescence my contaminate the cubes around that do not necessarily express the same gene see figure \ref{fig:cubeserrors}A. The second type of error is introduced by the choice of 3 \microm{3} cubes. As they are smaller than the average cell, some cubes will fall on areas that may be artificially empty. Indeed, transcription in the cells mainly happens in the nucleus, mRNA then travel in the cytoplasm to be translated but they are not evenly distributed across the cell, in particular for some large cells, parts of the cytoplasm may record no expression in a cell that actually contains a lot of transcripts, see figure \ref{fig:cubeserrors}B.\\
  
  Hence the data will tend to exhibit spatial discontinuity and inconsistency. With that fact in mind, any method that we develop using this data, will have to take into account this spatial discontinuity and try as much as possible to smooth over those potential expression gaps.\\
  
    \begin{figure}[h]
\centerline{\includegraphics[width=0.9\linewidth]{gfx/chapter2/cubeserrors.png}}
\caption{Errors introduced by the ``cube" cell model. Path A shows how regions with highly expressed genes can introduce errors through light contamination. Path B shows how some ubes may appear artificially void of expression because of the not even distribution of transcripts inside the cytoplasm especially for large cells.}\label{fig:cubeserrors}
	\end{figure}
	
	However, even with this simple cell model the data generated by \cite{Tomer10} is highly valuable. Indeed, not only does this dataset give a snapshot of gene expression for 169 genes in the full brain of \platy{}, it also attaches spatial information to each data point.\\

\section{Singe cell RNA sequencing, building a map of the full transcriptome}\label{sec:single_cell_rnaseq}
  \subsection{Sequencing single cell RNA contents}
	The scale shift from tissue to single cell is harder to achieve in the case of RNA-seq. As described in the previous section \ref{sec:gene_expression_lab}, an important factor for the success of RNA-seq assays is the starting input quantity of RNA to be sequenced. Taking mammalian cells as a reference, the quantity of RNA depends a lot on the cell type considered and can vary between 10 and 30 pg per cell, only 2\% of which is mRNA \cite{iscove02,islam11}. With such a small input quantity, distinguishing biological variation between different cells from the technical variation linked to cDNA amplification protocols has long been impossible to achieve.\\

	However, with the creation of new protocols \cite{ramskold12,tang09}, and the rise of microfluidics to faciliate the extraction and sequencing of single cells \cite{ozsolak10}, the last couple of years have seen a dramatic increase in the number of single cell RNA-seq based studies \cite{islam13,marinov13,yan13,staahlberg13,deng14}. However, challenges are yet to be overcome to be able to analyse further complex tissues from whole transcriptomes obtained from single cell RNA-seq, one of which is the loss of spatial reference induced by the current protocols.

  \subsection{Mapping back gene expression to a spatial reference}

	Single cell RNA-seq achieves to capture a snapshot of the entire transcriptome of a given cell at a given point in time. However, to analyse cells from a complex tissue, current protocols require that the tissue is reduced to a suspension of single independent cells. This prevents from keeping track of any spatial information about the cells. Hence, when analysing single cell RNA-seq data from a complex tissue, we need to be able to map back every cell to its original location.\\ 
	
	In order to achieve this back-mapping, a reference is needed. This reference should consist in an independent assay where gene expression in the considered tissue is defined for enough genes at a spatially small enough resolution to find for each sequenced cell, if not the exact original location of the cell, at least a restricted region of the tissue from which the sequenced cell originated with a high probability.\\
	
	Fortunately, in-situ hybridization assays provide exactly this type of data and we will present in the last section of this chapter \ref{sec:back_mapping_platy} a methodological proof-of-concept of this back-mapping in the brain of \platy{} with 72 sequenced single cells.

\section{About the quantitative trait of single cell expression data}\label{sec:quantitative_single_cell}
  \subsection{Light contamination in in-situ hybridization data}
  The fluorescence value obtained from in-situ hybridization assays can be considered as quantitative \cite{dorresteijn90}. Indeed, the light intensity emitted by every cell in the considered tissue is correlated with the number of RNA fragments present in the cell as each fragment bound to a probe is an independent source of emission and the probes are hybridized in the cells in large excess. This means that if the targeted gene is highly expressed in a cell, there will be more sources of emission, thus making the overall light intensity captured on this area higher than in a cell expressing the gene at a low level. \\
  
  As mentioned in a previous section \ref{sec:single_cell_insitu}, in-situ hybridization assays at the single cell level are prone to punctual errors due to the cell model. One of the culprit for those errors, as shown on figure \ref{fig:cubeserrors}B is the phenomenon of light contamination. When a large group of neighbouring cells express the same gene, because of the additivity of light intensity mentioned above, even though the cells express the gene at the same rate, cells surrounded by a lot of other cells expressing the same gene will have an abnormally high light intensity readings due to light contamination from the adjacent areas. As a result, when considering an hypothetical circular portion of tissue where a gene is monotonously expressed, the recorded light intensity will show a gradient with the maximum localized on the circle's centre.\\
  
   \begin{figure}[h]
\centerline{\includegraphics[width=\linewidth]{gfx/chapter2/whybina.png}}
\caption{{\bf Light contamination on in in-situ hybridization luminescence data seen on the example of gene Ascl.} Panel A shows the raw fluorescent microscopy capture of the gene's expression for one layer in the brain of \platy{}. Panel B shows the light intensity measured along the red line in panel A. Because of the small scale of study, cells surrounded by other cells expressing the same gene will have a higher intensity values because of nearby light contamination. Even though there might be an actual gradient in the expression level bewteen the cell in the middle and the others there is no option to separate it from the light contamination component.}\label{fig:why_binarize}
	\end{figure}
  
  As shown on figure \ref{fig:why_binarize}, we can confirm that light contamination issue on the in-situ hybridization data in \platy{}'s brain. In that context, and because of the single cell scale of our study, considering the in-situ hybridization data as quantitative may have introduced significant errors. In order to avoid this light contamination bias we decided to transform the quantitative data set into a binary data set where for a given ``cube", genes are simply expressed or not. The binarization method is described in the following section \ref{sec:binarizing}.
  


  \subsection{Technical noise in single cell RNA-seq data}
  Single cell RNA-seq is also prone to high levels of noise. This technical noise is caused by the minute amounts of starting RNA material. A study lead by Philip Brennecke1, Simon Anders and Jong Kyoung Kim \cite{brennecke13}, proposes a statistical method to overcome this high noise level and distinguish between biological variation and technical variation in the gene expression levels.\\
  
  To illustrate the dramatic increase in noise level, they used series of dilution assays, reducing step by step (5000 pg, 500 pg, 50 pg, 10 pg) the input quantity of RNA fragments extracted from \species{Arabidobsis thaliana} with two technical replicates each time using the Tang protocol \cite{tang09}. The authors of the study let us analyse this data, and we were able after normalizing by the size factor using the Bioconductor package DESeq \cite{anders10} to generate the scatter plots shown on figure \ref{fig:dilutions}. \\
  
  It is clear from these dilution assays that the noise level is correlated with the input quantity. Even though highly expressed genes are consistently well quantified even with 10 pg input material, for most of the genes, with less than 50 pg input RNA it seems dangerous to assume the results of single cell RNA-seq as quantitative with the current technological capabilities.
  
\begin{figure}[h]
        \myfloatalign
        \subfloat[500 0pg input RNA.]
        {\includegraphics[width=.45\linewidth]{gfx/chapter2/5000pg.png}} \quad
        \subfloat[500 pg input RNA.]
        {\includegraphics[width=.45\linewidth]{gfx/chapter2/500pg.png}} \\
        \subfloat[50 pg input RNA.]
        {\includegraphics[width=.45\linewidth]{gfx/chapter2/50pg.png}} \quad
        \subfloat[10 pg input RNA.]
        {\includegraphics[width=.45\linewidth]{gfx/chapter2/10pg.png}}
        \caption{Dilution series of total A. thaliana RNA}\label{fig:dilutions}
\end{figure}

	\subsection{Conclusions}
	The paragraphs above have shown that neither in-situ hybridization nor RNA-seq data can be safely assumed as quantitative when the scale is lowered at the single cell level. To avoid nonsensical conclusions linked to the noise level in the rest of the study, a solution was to binarize any single cell dataset that we were dealing with. However, binarization is not a trivial problem as discussed in the following section.



\section{Binarizing gene expression datasets}\label{sec:binarizing}
  \subsection{Binarizing in-situ hybrdization datasets}
	As shown in Figure \ref{fig:why_binarize} and discussed in the previous section \ref{sec:quantitative_single_cell}, we decided to avoid the various problems linked to light contamination by transforming the ``quantitative" fluorescence information into binary data. In other words, if $S$ is the set of all ``cubes" in the brain, $M$ the set of all the considered genes and $y_{i,m}$ the value retrieved from the in-situ hybridization data for ``cube" $i \in S$ and gene $m \in M$, then  $y_{i,m} = 1$ if gene $m$ is expressed at site $i$, $y_{i,m} = 0$ otherwise. The binarization process itself is not trivial. Indeed, defining the light intensity threshold above which a gene is considered expressed is a complicated problem, especially for noisy data.\\

	Looking at the density of intensities across all the ``cubes" for each gene, we found two very different scenarios: some densities were separated into two clear peaks, making the threshold easy to find while others exhibited a single peak making it hard to choose a clear cut value as shown in figure \ref{fig:densities_bina}. After trying different thresholding methods based on those densities, we found that the resulting binary expression was not satisfying for a large number of genes. Considering that this binarized dataset will be the cornerstone of the work presented in this thesis, it was very important to achieve a high confidence thresholding. Given the small number of genes studied (169), and the collaboration with a team of biologist working specifically on \platyfull{}'s brain, we decided to opt for a manual approach to thresholding. Indeed, by going through the 169 genes one by one, we adjusted the threshold manually until the resulting binarized expression patter corresponded perfectly to 1) the fluorescent stack images from in-situ hybridization data; 2) the biologically known expression patterns in the brain of \platy{} validated by the biologists.\\
	
	\begin{figure}[h]
\centerline{\includegraphics[width=\linewidth]{gfx/chapter2/densities_bina.png}}
\caption{Light intensity densities found for genes rOpsin and PRDM8.H92 accross \platy{}'s whole brain on a logaritmic scale. $N$ for each graph is the number of ``cubes" in the dataset where the fluorescence value is higher than $0$. On the one hand, the log density shows two clear peaks for rOpsin, making the choice of a expression threshold easy. PRDM8.H92 on the other hand does not display such a clear cut threshold.}\label{fig:densities_bina}
	\end{figure}
	
	This method resulted in a high confidence binarized dataset for $86$ genes. Several reasons explain why $83$ genes out of the starting $169$ were removed from the dataset. For some of the genes no good threshold could be found, this was due to high noise level in the in-situ hybridization images. Other images suffered from experimental errors resulting in blurred and unexploitable expression patterns. Finally some images where polluted by a well known experimental artefact linked to fluorescent microscopy imaging.\\
	
	Although the aforementioned method resulted in a high quality binary dataset, it has been possible only because the number of genes considered was small. This will not be the case when dealing with RNA-seq data.

  \subsection{Binarizing whole transcriptomes}
  
  When dealing with whole transcriptomes, manually finding thresholds to binarize gene expression data is no longer a valid option due to the high number of genes considered. An automated method is thus required. As we did not have access to a large single cell RNA-seq dataset to test the methods presented here, we will discuss possible ways to binarize single cell RNA-seq data, presenting some results from a small number (72) of sequenced cells in the brain of \platy{}, containing the count number for 169 genes (see next section \ref{sec:back_mapping_platy} of a detailed presentation of this data).\\
  
  A naive approach would be to simply consider that as long as one RNA fragment mapped to a particular gene has been found in a cell, the gene is considered as expressed. Although such a method would be justifiable in the case of a perfect dataset, with no noise or errors, as discussed above \ref{sec:quantitative_single_cell} in the case of single cell RNA-seq the noise level generated by the currently available sequencing technologies would prove too high to rely simply on this method. However, as a first approach on our dataset, on figure \ref{fig:hist_rna} see as predicted a very dominant peak for the value $0$. The problem remains that for very small count numbers it seems dangerous to set the gene as expressed.\\
  
\begin{figure}[h]
        \myfloatalign
        \subfloat[Histogram showing the frequencies of count values over 72 sequenced cells, with the fragments mapped to 169 genes.]
        {\label{fig:hist_rna}
        \includegraphics[width=.45\linewidth]{gfx/chapter2/hist_rna.png}} \quad
        \subfloat[Density plot for the count values over $0$ in the single cell sequencing dataset]
        {\label{fig:density_rna}%
         \includegraphics[width=.45\linewidth]{gfx/chapter2/density_rna.png}}
        \caption{Thresholding RNA sequencing data for \platy{}}\label{fig:platynereis_single_cell}
\end{figure}
  
  Another option would be to find a global threshold over the complete dataset. The threshold $T>0$ would represent the count number of RNA fragments for a particular gene and a particular cell needed to consider the gene as expressed. $T$ could be inferred from the count density over all the genes and all the cells. The expected result would be a 2 peaks density curve, one peak would correspond to the non expressed count values, the second to expressed genes. The binary threshold would then be set between the first and second peak. Although more precise than the previous method, binarizing in such a manner may lead to numerous errors. Indeed, the underlying assumption behind this method is that all genes behave in a similar way. As figure \ref{fig:density_rna} shows, if a 2 peak behaviour is indeed present, the cut is not extremely clear and an important portion of count numbers actually fall in between the two peaks. This is due to the fact that all expressed genes are not expressed in the same way, some lowly some highly, which has a tendency to flatten the density curve making this thresholding method, if better, still not 100\% reliable.\\
  
  The more suitable approach to this thresholding problem, would be to compute one threshold per gene based on the density curve for every gene accross all cells. However, with 72 cells into consideration, considering the sparse nature of the count data, we cannot show any results with this method on our dataset. We believe however, that one threshold per gene would prove a big improvement over the previously mentioned thresholding methods providing sufficient number of data points per gene.

\section{Preliminary results on mapping single cell RNA-seq data in from Platynereis' brain}\label{sec:back_mapping_platy}
  \subsection{Single cell RNA-seq in Platynereis' brain}
  
    	Collaborations with the Kaia Achim in the Detlev lab in EMBL have provided us with a unique RNA-seq dataset of 72 single cells from \platy{}'s 48hpf developing brain. 
    	
	Experimentally, the done work consisted in setting up \platy{} batches, picking up 50-100 individuals at 48hpf. These were washed in Ca, Mg - free sea water and incubated in a mixture of pronase (breaks extracellular matrix) and thioglycolate (helps to break the chorion). After this treatment, the trunks and epispheres were separated. 40-60 epispheres were then picked out, transferred to Phosphate buffered saline (PBS) and then incubated for 1 minute in PBS containing collagenase to break more extracellular matrix. After two PBS washing, the cells were dissociated by pipetting up and down then washed again 1 ml of 1xPBS and concentrated by centrifuging (1 min, 1000 rpm). Cells were re-suspended in 20 microliters of 1xPBS, of which 5 microliters could be loaded on the capture chip.\\

	Fluidigm's C1 Single-Cell Auto Prep System instrument with the Fluidigm Single-Cell Auto Prep IFC chip optimized for 10-17 micron cells were used. The reverse transcription was performed using Clontech SMARTer Ultra Low Input RNA Kit and for on-chip PCR the Clontech ADVANTAGE-2 PCR kit. Sequencing libraries were prepared using Nextera DNA Sample Preparation kit from Illumina.\\

	With two chips and a capture rate of 65\%, 72 libraries were sequenced including 11 cells from first chip, 35 live single cells, 17 dead single cells, 3 tubes from 2 cells, one with 4 cells, and 3 unsure ones from the second chip resulting in 72 raw reads files.\\	
  	
	Of course those results do not include the spatial localization of the cells as the protocol requires the separation of the coherent tissue into a cell suspension. As a crucial point in any dowstream analysis, we need to be able to map back the single cells to their original location in the brain. To that end, we took advantage of the spatially localized in-situ hybridization described in the previous section.\\

  \subsection{Mapping back RNA-seq data back to PrimR in-situ hybridization assays}
  	We started by mapping the RNA-seq reads to the 169 reference genes composing the in-situ hybridization data using Bowtie.\todo{cite nuno pipline}. The resulting dataset is the count number for each of the 169 genes in the 71 cells sequenced. In order to map back to the in-situ hybridization data, our approach consisted in extracting the genes that were the most specifically expressed for each sequenced cell, then compare this specific fingerprint to the in-situ 3D data in order to isolate the regions of the brain where those specific genes are co-expressed.\\
	
	Given the set of 169 considered genes $M$, and the set of 72 cells $C$, with the read counts matrix $D$ of size $M\times C$ we can compute for each cell and each gene, the expression specificity ratio $r_{m,c}$ defined as :
\begin{eqnarray*}
	r_{m,c} &= \frac{D_{m,c}.\left\|M\right\|}{\sum_{a \in C}D_{m,a}}
\end{eqnarray*}
By looking for each cell, at the genes with the highest ratio we can define the what genes are the most specifically expressed in  this cell compared to the other cells sequenced. On the one hand, this mapping method has the inconvenient of using the average expression level across all considered cells to compute the ratio $r$. This means that the mapping quality of each cell will depend on the overall sequencing quality. Furthermore, this method performance relies on the assumption that we do have in fact a collection of cells from different cell types, given the experimental protocol described above, this seems to be an acceptable hypothesis. On the other hand, this mapping method has the advantage of not being sensible to technical noise in the RNA-seq protocol, providing the technical noise between cells remains at a constant level. This explains the use of the read counts in a quantitative way and not a binarized dataset.\\

	The goals of this study were to validate the protocol used in order to obtain single cell RNA-seq results in \platy{}'s brain and to establish a methodological proof-of-concept on spatially mapping RNA-seq results onto in-situ hybridization data. We will present here a few examples of sequenced cells, their most specifically expressed genes and their resulting potential original location in the brain as well as the probable cell type they belong to.\\
	
	In table \ref{tab:rna_seq_representative_genes} are shown the most specific genes for four cells sequenced. With that list of genes that are supposed to be the most representative ones for the four cells considered, we visualized in 3D the areas where those genes are co-expressed in the brain of \platy{} according to the in-situ hybridization data. A snapshot of this visualization is shown on figure \ref{fig:cell_localization}. In every case, simply looking at the three most representative genes seems to allow a clear localization of the sequenced cells. Of course this mapping is not at the single cell level, but having an idea of the tissue every cell originated from is already a nice proof-of-concept.\\
	
	From the most specific genes to each cell and their potential localization, we can hypothesize from previous biological studies the cell type of each sequenced cell. As shown in table \ref{tab:rna_seq_representative_genes}, for cell ``X2C911L" the most specifically expressed gene ``Emx" has been used as a reference gene to localize the mushroom bodies, an hypothesis which is compatible with the co-expression of ``CALM.R29" and ``Dach" \cite{Tomer10}. Cell ``X2C521L" expresses Wnt8 very specifically, a gene shown to be linked to lateral brain development. Cell ``X2C61L" can be easily classified as a developing neuron. Indeed both VACht (Vesicular acetylcholine transporter) and ChaT (Choline acetyltransferase) are genes coding for enzymes interacting with the neurotransmitter acetylcholine. Finally cell ``X2C241S" displays the specific expression of the gene ``Mitf", one of \platy{} most studied gene and expressed solely in the developing adult eyes \cite{kozmik08,guy08}.
	
\begin{table}
    \myfloatalign
  \begin{tabularx}{\textwidth}{X|X|X|X} \toprule
    \tableheadline{X2C911L} & \tableheadline{X2C521L} & \tableheadline{X2C61L} & \tableheadline{X2C241S} \\ \midrule
    Emx & Wnt8 &  VAChT & Mitf\\
    CALM.R29 & HEN1-Y61 & ChaT & Otx\\
	Dach & Gsx & LYamide & Tolloid-Y68\\
    
\midrule
	Mushroom body & Developing lateral brain & Differentiated neural tissue & Adult eye\\
    \bottomrule
  \end{tabularx}
  \caption{Top 3 most specific genes for 4 sequenced cells and the potential tissue they belong to. The resulting localization of those four cells infered from the in-situ hybridization data are shown on figure \ref{fig:cell_localization}.}\label{tab:rna_seq_representative_genes}
\end{table}

\begin{figure}[h]
        \myfloatalign
        \subfloat[X2C911L: Emx, CALM.R29, Dach]
        {\includegraphics[width=.45\linewidth]{gfx/chapter2/cell1.png}} \quad
        \subfloat[X2C521L: Wnt8, HEN1-Y61, Gsx]
        {\includegraphics[width=.45\linewidth]{gfx/chapter2/cell2.png}} \\
        \subfloat[X2C61L: VAChT, ChaT, LYamide]
        {\includegraphics[width=.45\linewidth]{gfx/chapter2/cell3.png}} \quad
        \subfloat[X2C241S: Mitf, Otx, Tolloid-Y68]
        {\includegraphics[width=.45\linewidth]{gfx/chapter2/cell4.png}}
        \caption{Regions defined by the expression overlap of the top 3 scoring genes in \cite{Tomer10} binarized in-situ hybridization data. The red colour shows the co-expression of the three considered genes, the blue areas are those where one or more of the three considrered genes are expressed but not all, in grey are the areas where none of the considered genes are expressed. The 4 figures are from a apical view with the dorsal side on top.}\label{fig:cell_localization}
\end{figure}


\section{Conclusions}
In this chapter we presented how the scale of gene expression studies has shifted from the tissue level to the single cell level. For two experimental protocols we have described how such data can be obtained and why at the time of writing, it is still not safe to assume the single cell datasets as quantitative. To avoid that problem, turning those datasets into binary gene expression is an attractive solution. However the binarization process is not trivial and we have presented ways to obtain a high confidence dataset.\\

One big advantage of in-situ hybridization assays is the fact that the spatial information stays attached to each gene expression fingerprint. Using this information, we were able to replace single cell RNA-seq data in a spatial context, using the most specifically expressed genes in every cell. To our best knowledge an a posteriori localization of single cell RNA-seq data had never been presented before.\\

As mentioned previously \ref{ch:background}, if the ability to study the heterogeneity of cell populations at the single cell level is offers incredible possibilities for the future of developmental biology and of cancer research, the development of new statistical methods adapted to this single cell scale allowing conclusions to be drawn at the tissue level is crucial.\\

The work presented thereafter was done to answer simple but important questions: can we define and localize known functional tissues of a complex organ like the brain from single gene expression data? Can we detect unknown regions in such a complex tissue and finally can we hypothesize the functional role of those unknown regions based on the single cell expression data?


	
	



%*****************************************
%*****************************************
%*****************************************
%*****************************************
%*****************************************
