%
\chapter{Capturing gene expression in \platyfull{}'s brain}\label{ch:background}
%
\section{Platynereis dumerilii, an ideal organism of brain development studies}\label{sec:platynereis}
     \subsection{General description}
     \platyfull{} is a marine annelid of the class Polychaeta, it has been established as one of the main marine animal models in the fields of evolutionary, developmental biology as well as ecology and toxicology and neurobiology \cite{hutchinson95,tessmar03,hardege99,dorresteijn90,fischer04,Fischer10}. As a member of the bilaterians \platy{} has a defined bilateral symmetry.\\
     
     \platy{} populates shallow (no more than 3m) hard ocean floors around the world. It is commonly found in the Mediterranean sea, the north Atlantic coast of Europe as well as in the shallow seas surrounding Sri Lanka, Java and the Philippines. Eggs, embryos and larvae are roughly 160 $\mu$m while the adults can measure up to 6cm in length.
     
     
\begin{figure}[bth]
        \myfloatalign
        \subfloat[Larval form of \platy{}. Image: MPI for Developmental Biology.]
        {\label{fig:platynereis_larvae}
        \includegraphics[width=.45\linewidth]{gfx/chapter1/platynereis_larva.jpg}} \quad
        \subfloat[Adult \platy{}. Image: Arendt group, EMBL]
        {\label{fig:platynereis_adult}%
         \includegraphics[width=.45\linewidth]{gfx/chapter1/platynereis_adult.jpg}}
        \caption{\platyfull{}'s larva and adult forms.}\label{fig:platynereis}
\end{figure}
     
     They are several reasons why \platy{} has been chosen as a model by numerous laboratories. Indeed, evolutionary wise, \platy{} shows several interesting characteristics. It belongs to the lophotrochozoan taxon of the bilaterians as opposed to most of the well established model animals which either belong to the ecdysozoans (\species{Caenorhabditis elegans}, \species{Drosophila melanogaster}) or the deuterostomes (mouse, human). Lophotrochozoans being extremely under represented, \platy{} as a model organism is essential to be able to use comparative approaches on bilaterians \cite{Fischer10}.\\
     
     \platy{} also exhibits an exceptionally slow evolutionary lineage. It has even been described as a ``living fossil" for that reason \cite{Fischer10}. Therefore, the numerous ancestral developmental characteristics exhibited by \platy{} translate into an image of the common past of all bilaterians. For example, an interesting example described in \cite{denes07,tessmar07} is the conserved molecular topography of the genes responsible for the development of the central nervous system between \platy{} and all vertebrates. This slow evolutionary rate confers to \platy{} the advantage of being a link between fast evolving models like \species{Drosophila} and vertebrates.\\
     
     In terms of practicality, \platy{} can easily be kept and bred in captivity producing offspring throughout the year \cite{fischer04}. The behavioural characteristics of \platy{}'s mating ritual have been well studied. The ``nuptial dance" happens on the water surface. Male and female respectively release the sperm and eggs synchronously. This activity is synchronized by pheromones released into the water \cite{zeeck98}. Over 2000 individuals can be produced within a single batch. Every new individual will undergo embryonic then larval development before reaching \platy{}'s adult form.\\

 
     \subsection{Larval development}
    Similarly to the other polychaetes, the larval development of \platy{} can be decomposed into three main anatomical stages: the trochophore, the metotrochophore and the nectochaete. The trochophore is spherical and moves thanks to equatorial belt of ciliated cells as well as an apical organ displaying a ciliary tuft \cite{rouse99,nielsen04} as seen on figure \ref{fig:platynereis_larvae} and schematically on figure \ref{fig:platynereis_larvae_scheme}. the metotrochophore stage is characterized by the development of a slightly elongated segmented trunk compared to that of the trochophore \cite{hacker98}. The next stage is the nectochaete larvae that resembles the adult (figure \ref{fig:platynereis_adult}) in most of the traits especially with parapodial appendages used for swimming and crawling \cite{hacker98}. This traditional subdivision has been applied to \platy{} \cite{hauenschild69}.\\
    
\begin{figure}[bth]
\begin{center}
  \includegraphics[width=0.6\linewidth]{gfx/chapter1/larvae48hpf.png}
\end{center}
  \caption{\platyfull{}'s larvea development at 48hpf or late trochophore. Striped in red is indicated the area which forms the developing brain of the larvae.}
  \label{fig:platynereis_larvae_scheme}
\end{figure}
    
    Aside from this purely anatomical description, an additional staging system exists and has become the norm for current studies. The development is measured in \textit{hours post fertilization} (hpf) at $18^{\circ}C$.
    
    A key factor making \platy{} such an interesting model to work with is the fact that after fertilization, the $\approx 2000$ larva will start developing at the exact same time, in a synchronous fashion. Furthermore, the larval development of \platy{} follows a very stereotypical pattern with very little variation from one individual to the other and even between batches provided the temperature is kept constant \cite{fischer04,dorresteijn90}. An example showing the similarity between individuals during development can be seen on figure \ref{fig:brain_comparison}. this is a very important feature as it allows biologists to repeat experiments on several individuals at a very close developmental stage even if they are from different batches.\\
    
\begin{figure}[bth]
  \includegraphics[width=\linewidth]{gfx/chapter1/brain_comparison.png}
  \caption{\platyfull{}'s stereotypical and synchronous development. In green and red are two different \platy{} individuals' with the same gene expression being highlighted. They show extremely similar patterns of development.}
  \label{fig:brain_comparison}
\end{figure}

	 Describing the entire development of \platy{} does not fall within the scope of this thesis. Indeed, I will only be interested in the brain of \platy{}'s larvae at 48hpf. Therefore, it is important to have an anatomical idea of what the brain looks like at this time in development and what inherent characteristics will be the most interesting to investigate.

\section{Gene expression in Platynereis' developing brain}\label{sec:gene_expression_background}
     \subsection{Platynereis' nervous development until 48hpf}
     The main purpose of this thesis is not to fully understand the patterns of development in \platy{}'s larval brain. Therefore I will only give a brief summary of what the main component of the brain are at 48hpf, the time point I will be interested in in the next chapters. \platy{}'s larval brain development is detailed in \cite{Fischer10}.\\
     From the early trochophore (24-26hpf) neural system development starts taking place. The apical ganglion forms at the apical tuft. It contains one sertonergic cell and a few neurons linked to the nerve of the ciliary band of the larva called the prototroch (see figure \ref{fig:platynereis_larvae_scheme}). This allows the first movements of the larve thanks to the cialited cells of the prototroch.\\
     The mid-trochophore (26-40 hpf) sees the formation of the first cerebral commissure, it is a band of nerves interconnecting the ventral nerve cord and the brain. This trait is a typical feature of annelids' brains. During this phase the apical ganglion becomes bigger with three more serotonergic cells.\\
     The late trochophore (40-48hpf) sees the formation of the second commissure in the ventral nerve cord. It is at the end of this stage that the tissues of the brain become more complex with a notable increase in the number of neutrites \cite{Fischer10}.\\
     The data used in the rest of this thesis will not encapsulate the whole larvae, just the developing brain (see figure \ref{fig:platynereis_larvae_scheme}) thus excluding the ventral part of the nervous system. The best studied areas of the developing brain are the larval eyes, the developing adult eyes and the apical organ on the dorsal side. On the ventral side are located the mushroom bodies, a pair of structures that are known to play a role in olfactory learning and memory in insects and annelids \cite{Tomer10}.\\
     
     Even at this early stage in a relatively ``simple" organism, the brain quickly becomes an extremely complex tissue. Cell types diverge and functional areas are formed. Before trying to understand more about \platy{}'s brain organization, it is interesting to ask the more general question of how complex tissues such as the brain are defined spatially.
     
          \subsection{Spatial organization of complex biological tissues like the brain}
     This section does not intend to demonstrate any specificity in the brain of \platy{}, it is meant to ask some of the fundamental questions that intrinsically motivate the work presented in the rest of this thesis. Complex tissues, the obvious example of which is the brain, could be viewed as an interconnected mosaic of cells having different functions, working together to achieve the global function of the organ.\\
     
     When looking closely at this mosaic of cells, it is easy to observe that the spatial organisation is not random. Indeed, cells that serve the same function will often be close to each other, thus defining functional tissues. However, the spatial coherency of those tissues is not necessarily always the same. Some cell types may consist of cells that are scattered inside another more spatially coherent tissue. To illustrate that fact, an interesting example is the difference between the spatial coherency of cells forming the neuronal tissue in the brain and cells forming a well defined region in the brain like an exocrine gland. When asking the question: ``is it likely that this particular cell is fully surrounded by cells belonging to the same cell type?", the extensions created by the axons of neurones will decrease this probability. Indeed, axons will grow through other types of tissues to reach their destination \cite{bartlett84,colello90}, making the overall spatial coherency of neural tissues smaller than very well spatially defined tissues.\\
     
     When trying to analyse the full structure of the brain with an automated method, keeping in mind that fact could prove important to improve the results. This fact and its consequences on the work presented in this method are further discussed in section \todo{cite section spatial clustering}.\\
     
     So far, organs and cell types have been defined by their anatomical traits. But as mentioned in the introduction \ref{ch:introduction} the functional heterogeneity of complex tissues goes further that simple anatomical traits. Instead interesting traits are those that fundamentally define how cells are functioning.
     
     \subsection{Generalities about gene expression and development}  
     Throughout this thesis the term cell will be refer to eukaryotic cells and more specifically those of multicellular organisms. Every cell in a complex organism possesses the same genome, that is, the sum of all the genetic information contained in the cell (nucleus and other compartments). This fundamental homogeneity is in plain contradiction with the heterogeneity observed anatomically. If every cell has the exact same DNA, where does the great variability between cell types come from (what makes a neurone become a neurone and not a pancreatic cell)? Answering this sort of questions defines the field of developmental biology.\\
     
     The short and rather complete answer to any developmental biology question actually is: same genome but different pattern of gene expression. As gene expression is the central, most important, most studied cellular activity, gene expression is indeed the common denominator of life, as large parts of the mechanisms making up gene expression are actually shared by every living creature known to man.\\

     Of course to understand what gene expression is, the notion of genes must first be defined. The precise definition of a gene is still controversial. The concept of a ``factor that conveys traits from parents to offspring" was laid by Gregor Mendel in 1866 \cite{mendel66} when the accepted theory at the time was based on blending inheritance where the traits of the parents appeared mixed in the offspring following a continuous gradient. The most recently published definition of a gene followed the publication of the ENCODE project \cite{feingold04}. It states that a gene is ``a union of genomic sequences encoding a coherent set of potentially overlapping functional products."\\

	Gene expression is the way cells express their genes. Expression of a gene is the process of transcribing the DNA of that particular gene. It is interesting to note that there are several ways to look at gene expression. Indeed in a cell or tissue, at a given time point it is possible to look whether a gene is expressed or not (binary expression) or how much a certain gene is expressed (quantitative expression). The product of gene expression is RNA molecules.\\
	
	Most RNA molecules are translated into proteins that can have very different purposes. Some will serve directly in the cellular life as functional/structural agents (elements of the ATP synthase for example \cite{boyer97}), others will be excreted by the cell and will serve a purpose at the scale of the organism \cite{kaiser84} others called transcription factors will have a regulatory effect on gene expression \cite{mitchell89}. In other terms the expression of gene $G_a$, coding for protein $P_a$ might activate, accelerate, inactivate or decelerate the expression of gene $G_b$ and potentially others. This outlines the complex interdependent regulatory system that is gene expression, see figure \ref{fig:cells}. For precise examples gene regulation see \cite{gossen92, shinozaki03,fuqua01,balmer02}.\\
	
\begin{figure}[bth]
        \myfloatalign
        \subfloat[Undifferentiated cell with a circular shape.]
        {\label{fig:undiff_cell}
        \includegraphics[width=.45\linewidth]{gfx/chapter1/cell.png}} \quad
        \subfloat[Differentiated neuron, with typical axons development.]
        {\label{fig:diff_cell}%
         \includegraphics[width=.45\linewidth]{gfx/chapter1/neurone.png}}
        \caption{Cell types anatomical heterogeneity, gene expression and protein translation and gene regulatory networks. The schematics shows that genes in the DNA are transcribed to RNA molecules that are further translated outside the nucleus into proteins. Those proteins can serve various purposes inside the cell or come back to the nucleus to regulate gene expression.}\label{fig:cells}
\end{figure}
	
	During development, mechanisms exist that allow gene expression to become differential as the divisions occur. This is how the asymmetrical axis (dorso-ventral, and basal-apical) of the body are defined. The main mechanism involves chemical gradients. The first of these gradient has to come from the zygote which must contain some asymmetrically distributed chemical so that the first divisions lead to non identical cells. In the case of \platyfull{}, the body axis are defined  between 2hpf and 7hpf \cite{Fischer10}.\\
	
	As described, gene expression is the key factor during tissue development. The ability to study gene expression patterns has revolutionized the field of developmental biology. Technological innovation has been the main driving factor of this revolution. In the next section I will present two methods to capture gene expression.


\section{Capturing gene expression in the laboratory}\label{sec:gene_expression_lab}
     \subsection{In-situ hybridization assays}
     In-situ hybridization (ISH) is an experimental technique where the practitioner is able to determine in which cells of the tissue under study a particular RNA is present. As opposed to Southern blotting \cite{southern75}, ISH assays not only allow to know whether a gene is expressed or not, but also where in the tissue it is expressed. First proposed in 1969 by Pardue \cite{pardue69} and John \cite{john69} independently, in-situ hybridization (ISH) used radioactive tritium labelled probes on a photographic emulsion to reveal parts of the studied tissues where particular RNAs or DNA regions were present. With the development of fluorescent labelling techniques \cite{landegent84,pinkel88} allowing for faster, more sensitive and of course safer hybridization assays compared to radioactive probes \cite{swiger96}, Fluorescent in-situ hybridization (FiSH) quickly became the standard technique to study gene expression in the spatial context of the biological tissue. Importantly, using multiple fluorescent probes of different colours allowed the simultaneous localization of several RNA fragments in the tissues \cite{nederlof89}.
     

     \subsection{Building a image library of gene expression for Platynereis}
     During his PhD, Raju Tomer and other \todo{cite thesis} members of the Detlev Arendt lab in EMBL, used Fluorescent in-situ hybridization to create an image library of gene expression in the brain of \platy{}. He was able to record gene expression in the full brain at 48hpf for 169 genes. In practice each individual larvae was dissected to isolate the brain, which was then cut into thin slices and fixed. Each individual slice was then stained with two different fluorescent probes corresponding to two messenger RNAs (mRNA). One of the gene is considered a reference, as it is always hybridized in all the assays (the main reference gene used was Emx) alongside an other gene of interest, see figure \ref{fig:insitu}.\\
     
     As mentioned previously, the larval development of \platy{} is highly similar in every individual larvae. In the case of this study requiring a lot of different assays conducted each time a on different animal, the stereotypical development of \platy{} has proven essential. Indeed, having the same reference localized in all the assays has allowed Tomer to align all the other gene expression patterns onto this scaffold. The result is an image library of 169 gene expression patters in the full brain of \platy{} with a exploitable spatial reference that allows for a very precise mapping.\\
    
    \begin{figure}[bth]
\centerline{\includegraphics[width=0.9\linewidth]{gfx/chapter1/insitu.png}}
\caption{Fluorescent in-situ hybridization assays to create a 169 genes catalogue of gene expression in the brain of \platy{}. From the live tissue cut into thin fixed layers, every slice is stained with a reference gene and a gene of interest that will reveal the areas of expression under fluorescent microscopy. The process repeated 169 times for key genes in \platy{} development has been generated by \cite{Tomer10}}\label{fig:insitu}
	\end{figure}
	
	However useful and practical fluorescent in-situ hybridization may be, such assays are limited in terms of the quantity of gene one is able to study. Indeed, each individual larvae only provides the expression of two genes, one being the reference. Crucial developments in sequencing technologies have brought a way to study the expression of the whole transciptome landscape in a single assay, RNA sequencing.

     \subsection{RNA sequencing}
     Whole Transcriptome Shotgun Sequencing (WTSS) also called RNA sequencing (RNA-seq) \cite{morin08,wang09} has developed alongside Next Generation Sequencing (NGS) techniques used to retrieve the genome (DNA). Instead, when preparing the starting material, only the RNAs are extracted. Protein coding mRNA can further be selected, they are separated from the rest by targeting the polyadenylated 3' tail, specific to protein coding transcripts. Most current technique use magnetic beads to achieve this separation \cite{mortazavi08,morin08}.\\
     
    Once isolated from a population of cells, transcripts undergo fragmentation to obtain an average length of 200-300 nucleotides. The next step is the reverse transcription, which will create a complementary DNA (cDNA) library using viral reverse transcriptase enzymes. After amplification using quantitative Polymerase Chain Reaction (qPCR), the cDNA library is ready to be sequenced by NGS technology.\\
    
    This will generate a large dataset of small reads, that needs to be mapped back onto the reference genome of the considered species, providing this genome is available. In that case, the resulting dataset will reflect a snapshot of the whole transciptome in the studied cell population. However, in the case of \platy{}, this reference genome is not fully available yet, an alternative option is to map the reads back to a list of known gene sequences, for instance the 169 genes studied by \cite{Tomer10} (PrimR genes). The resulting dataset will represent a quantitative image of the considered genes in the cell population at one point in time.\\
    
    Because of technical limitations in this sequencing protocol, until very recently the starting quantity of RNA had to be relatively important (this issue is discussed further in \ref{sec:single_cell_rnaseq}). This is why most of the published RNA sequencing studies use a population of cell as a starting point. This however, means that the gene expression landscape obtained as an output will represent an averaged expression over all the cells used as an input.\\
    
    Importantly, when comparing RNA-seq the the previously described in-situ hybridization technique, if the methodological burden to analyse the expression of a lot of genes at the same time is greatly reduced, the spatial localisation of the cells is lost during the protocol.\\
    
\section{Conclusions}
     
     In this chapter I have presented \platyfull{} and the advantageous traits it exhibits for developmental biologists especially in the field of neural development. I have discussed the fact that anatomical traits are not sufficient to fully comprehend the deep heterogeneous patterns of functionalities inside a complex organ such as the brain. In order go push this understanding further an interest had to be taken in what defines the life of tissues and sub-tissues: gene expression. I have also described two methods that allow practitioners to capture gene expression from a biological tissue, and how an image library  of gene expression for 169 gene was generated by \cite{Tomer10} in the full brain of \platy{}.\\
     
	So far, the scale of study has been the tissue, or the sub-tissue. However, as mentioned in the introduction \ref{ch:introduction}, the heterogeneity of complex biological tissues does not stop at this scale of study. In fact, with a top-down approach looking at big tissues and then separating them in smaller sub-tissues until ``true" functional tissues are defined is an extremely complicated problem. A solution to this problem would be to reverse the approach from a top-down to a bottom-up mindset. This means reducing the scale of study to the smallest biological unit available, the single cell, define the heterogeneity of gene expression at the single cell level and go back up to the functional tissue level from there. Instead of a fragmentation problem, this becomes clustering problem, attaching single cells to a certain number of categories. In order to implement such an approach, single cell gene expression data is therefore needed.

%
%
%
%
%



